tests/Oryx.BuildImage.Tests/Node/NodeJsSampleAppsOtherTests.cs

```diff
@@ -672,6 +672,43 @@ public void BuildsNodeApp_AndDoesNotZipNodeModules_IfZipNodeModulesIsFalse()
                 result.GetDebugInfo());
         }
 
+        [Fact]
+        public void GeneratesScript_AndBuilds_UsingSuppliedPackageDir()
+        {
+            // Arrange
+            var volume = DockerVolume.CreateMirror(
+                Path.Combine(_hostSamplesDir, "nodejs", "monorepo-lerna-yarn"));
+            var appDir = volume.ContainerDir;
+            var appOutputDir = "/tmp/output";
+            var script = new ShellScriptBuilder()
+                .SetEnvironmentVariable(
+                    SdkStorageConstants.SdkStorageBaseUrlKeyName,
+                    SdkStorageConstants.DevSdkStorageBaseUrl)
+                .SetEnvironmentVariable(
+                    SettingsKeys.EnableNodeMonorepoBuild,
+                    true.ToString())
+                .AddBuildCommand($"{appDir} -i /tmp/int -o {appOutputDir} --package -p {NodePlatform.PackageDirectoryPropertyKey}=packages/app1")
+                .AddFileExistsCheck($"{appOutputDir}/packages/app1/app1-0.1.0.tgz")
+                .ToString();
+
+             // Act
+            var result = _dockerCli.Run(new DockerRunArguments
+            {
+                ImageId = Settings.BuildImageName,
+                Volumes = new List<DockerVolume> { volume },
+                CommandToExecuteOnRun = "/bin/bash",
+                CommandArguments = new[] { "-c", script }
+            });
+
+            // Assert
+            RunAsserts(
+                () =>
+                {
+                    Assert.True(result.IsSuccess);
```

Comment:
i think it will be good idea to assert on some other conditions as well ...  may be assert on this

> echo "Switching to package directory provided: '{{ PackageDirectory }}'..."